pkg/provider/aws/image.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package aws
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+)
+
+type ImageInfo struct {
+	ImageID      string
+	CreationDate string
+}
+
+type ByCreationDate []ImageInfo
+
+func (a ByCreationDate) Len() int           { return len(a) }
+func (a ByCreationDate) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a ByCreationDate) Less(i, j int) bool { return a[i].CreationDate > a[j].CreationDate }
+
+func (a *Client) getAMI() (string, error) {
+	var ami string
+
+	// Work around for AWS types
+	filterName := "name"
+	filterArchitecture := "architecture"
+	filterOwner := "owner-id"
+	awsOwner := "099720109477" // Canonical
```

Comment:
Since these are probably the properties that we would want to change going forward, we probably want a new type that HAS A client. These can be set at construction to find the required image.

Reply:
you mean, creating a NEW type on this file `image` to handle this `getAMI` func and related upcoming functions. and this type to share the `ec2` client?

Reply:
We now check if the user defined a specific owner id, if not we pass the default ones

Reply:
To be clear, this is probably only required when we start adding different conditions here.

Reply:
Yup, this is more a followup, right now we are hard coded to Ubuntu, once we start opening up to RPM based OS's this would require more logic. but as it stands now, I have tested x86 and arm, and it returns a valid image AMI id